### PR TITLE
Make earnings list month entries collapseable and change default of it for newly added widget

### PR DESCRIPTION
**Pull request by kimmerin**
Hi,

this change is a first step in the redesign of the earnings list widget as discussed in #4010. It makes month-entries "clickable" and leads to the collapse and expanse of the entries. This allows the widget to take up less space. I've also added a new configuration entry to control the default behavior (collapse all, expand current month, expand all). The deault behavior of newly added widget is "expand current month" that I regard as a useful default. If a widget has been added already, the previous "default" "expand all" is used.

### CODE REVIEW

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/earnings/EarningsListWidget.java [hunk 1]
```diff
@@ -275,14 +329,57 @@ public void update(Model model)
...
+            {
+                // empty, we're waiting for mouse-up
+            }
+
+            @Override
+            public void mouseDoubleClick(MouseEvent e)
+            {
+                // maybe another time with another function behind it
+            }
+        });
```
Comment (buchen) [inline name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/earnings/EarningsListWidget.java:382]: BTW, you can use also the adapters if we anyway implement only one method

```java
        label.addMouseListener(MouseListener.mouseUpAdapter(e -> {
            collapsed[mVal] = !collapsed[mVal];
            update();
        }));
```

**Comment by buchen** [comment]
I have adapted the pull request according to the comments and merged it. Thanks.

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/Messages.java [hunk 2]
```diff
@@ -1328,6 +1328,11 @@ public class Messages extends NLS
     public static String OptionDateIsInTheFuture;
     public static String OptionDateIsInThePast;
     public static String YearlyPerformanceHeatmapToolTip;
+
+    public static String ExpandAll;
+    public static String ExpandCurrentMonth;
+    public static String CollapseAll;
+    public static String LabelTreeExpansionConfig;
```
[the remark this answers is not in the document]
Comment (kimmerin) [reply name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/Messages.java:1335]: I'm adding these lines by hand but use a tool to edit the resource files

File: name.abuchen.portfolio/src/name/abuchen/portfolio/model/Dashboard.java [hunk 3]
```diff
@@ -60,6 +60,7 @@ public static final class Widget
         private String type;
         private String label;
         private Map<String, String> configuration;
+        private transient boolean newWidget;
```
[the remark this answers is not in the document]
Comment (kimmerin) [reply name.abuchen.portfolio/src/name/abuchen/portfolio/model/Dashboard.java:63]: I wasn't sure if this particular feature was useful for more than the widget I've worked on and in such a case I keep the change as small as possible and do a refactor at a later point in time if it turns out that it's useful for others as well.

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/messages_cs.properties [hunk 4]
```diff
@@ -212,6 +212,8 @@ ClientEditorLabelReturnsVolatility = V\u00FDnos / Volatilita
 
 ClientEditorPerformanceCalculation = Propo\u010Det
 
+ColumnCollapseAll = Sbalit v\u0161e
```
[the remark this answers is not in the document]
Comment (kimmerin) [reply name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/messages_cs.properties:215]: I'm using ResourceBundle Editor but in this particular case I've "lost" the changes when working with git (don't remember the details, though) so I've ended up adding the changes again by hand. Seems that I've done a copy/paste-error here.

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/messages.properties [hunk 5]
```diff
@@ -216,6 +216,8 @@ ClientEditorLabelReturnsVolatility = Return / Volatility
 
 ClientEditorPerformanceCalculation = Calculation
 
+CollapseAll = Collapse All
```
[the remark this answers is not in the document]
Comment (kimmerin) [reply name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/messages.properties:219]: I've assumed that and searched for them. Seems that I've missed them.

File: name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/earnings/EarningsListWidget.java (hunk 1 above)
Comment (kimmerin) [reply name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/earnings/EarningsListWidget.java:382]: I knew that there was something like this... ;-)

**Pull request closed without merging**